leshan-client-core/src/main/java/org/eclipse/leshan/client/observer/LwM2mClientObserver.java

```diff
@@ -0,0 +1,51 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Sierra Wireless and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *     Sierra Wireless - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.leshan.client.observer;
+
+import org.eclipse.leshan.ResponseCode;
+import org.eclipse.leshan.client.servers.DmServerInfo;
+import org.eclipse.leshan.client.servers.ServerInfo;
+
+
+/**
+ * Allow to observer the registration life cycle of a LwM2m client.
+ */
+public interface LwM2mClientObserver {
+
+    void bootstrapSucceed(ServerInfo bsserver);
+
+    void bootstrapFailed(ServerInfo bsserver, ResponseCode responseCode, String errorMessage);
+
+    void bootstrapTimeout(ServerInfo bsserver);
+
+    void registrationSucceed(DmServerInfo server, String registrationID);
+
+    void registrationFailed(DmServerInfo server, ResponseCode responseCode, String errorMessage);
+
+    void registrationTimeout(DmServerInfo server);
+
+    void updateSucceed(DmServerInfo server, String registrationID);
+
+    void updateFailed(DmServerInfo server, ResponseCode responseCode, String errorMessage);
+
+    void updateTimeout(DmServerInfo server);
+
+    void deregisterSucceed(DmServerInfo server, String registrationID);
+
+    void deregisterFailed(DmServerInfo server, ResponseCode responseCode, String errorMessage);
+
+    void deregisterTimeout(DmServerInfo server);
+}
```

Comment:
Most handlers use a naming pattern for methods following the `onSomeEvent` pattern. What do you think about using this here as well, e.g. `onBootstrapSucceeded` and `onRegistrationFailed` etc?